lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -3138,23 +3139,60 @@ bool TypeSystemSwiftTypeRef::IsTupleType(lldb::opaque_compiler_type_t type) {
   return false;
 }
 
+void TypeSystemSwiftTypeRef::DumpTypeDescription(
+    opaque_compiler_type_t type, lldb::DescriptionLevel level,
+    ExecutionContextScope *exe_scope) {
+  LLDB_SCOPED_TIMER();
+
+  StreamFile s(stdout, false);
+  DumpTypeDescription(type, &s, level, exe_scope);
+}
+
+void TypeSystemSwiftTypeRef::DumpTypeDescription(
+    opaque_compiler_type_t type, Stream *s, lldb::DescriptionLevel level,
+    ExecutionContextScope *exe_scope) {
+  LLDB_SCOPED_TIMER();
+  DumpTypeDescription(type, s, false, true, level, exe_scope);
+}
+
 void TypeSystemSwiftTypeRef::DumpTypeDescription(
     opaque_compiler_type_t type, bool print_help_if_available,
-    bool print_extensions_if_available, lldb::DescriptionLevel level) {
+    bool print_extensions_if_available, lldb::DescriptionLevel level,
+    ExecutionContextScope *exe_scope) {
   LLDB_SCOPED_TIMER();
-  if (auto *swift_ast_context = GetSwiftASTContext())
-    return swift_ast_context->DumpTypeDescription(
-        ReconstructType(type), print_help_if_available, print_help_if_available,
-        level);
+  StreamFile s(stdout, false);
+  DumpTypeDescription(type, &s, print_help_if_available,
+                      print_extensions_if_available, level, exe_scope);
 }
+
 void TypeSystemSwiftTypeRef::DumpTypeDescription(
     opaque_compiler_type_t type, Stream *s, bool print_help_if_available,
-    bool print_extensions_if_available, lldb::DescriptionLevel level) {
+    bool print_extensions_if_available, lldb::DescriptionLevel level,
+    ExecutionContextScope *exe_scope) {
   LLDB_SCOPED_TIMER();
-  if (auto *swift_ast_context = GetSwiftASTContext())
-    return swift_ast_context->DumpTypeDescription(
+  // Currently, we need an execution scope so we can access the runtime, which
+  // in turn owns the reflection context, which is used to read the typeref. If
+  // we were to decouple the reflection context from the runtime, we'd be able
+  // to read the typeref without needing an execution scope.
+  if (exe_scope) {
+    if (auto *runtime =
+            SwiftLanguageRuntime::Get(exe_scope->CalculateProcess())) {
+      const auto initial_written_bytes = s->GetWrittenBytes();
+      s->Printf("Typeref info:\n");
+      runtime->DumpTyperef({this, type}, this, GetSwiftASTContext(), s);
+      if (s->GetWrittenBytes() == initial_written_bytes)
+        s->Printf("<could not resolve type>");
+    }
+  }
+  // Also dump the swift ast context info, as this functions should not be in
+  // the hot path of printing frame variables, and it could be useful
+  // information.
+  if (auto *swift_ast_context = GetSwiftASTContext()) {
```

Comment:
@adrian-prantl I decided to dump both the typeref info + swift ast context, since this should give users more useful stuff to work with, and this function isn't very used anyway.

Reply:
I guess that's fine. It shouldn't be on anyone's critical path.